### PR TITLE
Added prettier-plugin for formatting glimmer component templates

### PR DESCRIPTION
**Pull request by wondersloth**
- Creates a prettier plugin to format `hbs` tagged template expressions with prettier's glimmer printer.
- Updated prettier to latest `2.3.1`
- Updated eslint to latest
- Decoupled prettier (formatting) from eslint.

### CODE REVIEW

**Comment by wondersloth** [comment]
Looking into build issues. Looks like my fixtures are being compiled.

**Comment by wondersloth** [comment]
Investigating some formatting issues. Inline hbs template do not prioritize double-quotes for the html. It should do this like html tagged template expressions and JSX.  Noting it also respects the `"singleQuote": true` option in `.prettierrc`. 

It also looks like the plugin is being run by the linter despite not being a published packaged. Changing .prettierrc to:
```diff
{
  "singleQuote": true,
  "trailingComma": "es5",
  "printWidth": 100,
+  "pluginSearchDirs": ["./node_modules"]
}
```

File: package.json [hunk 1]
```diff
@@ -13,15 +13,17 @@
   ],
   "scripts": {
     "build": "scripts/build.js",
+    "format:check": "prettier --check 'packages/**/*.{js,ts}'",
     "format": "prettier --write 'packages/**/*.{js,ts}'",
-    "lint": "eslint . --cache --ext .js,.ts",
+    "lint": "eslint . --ext .js,.ts",
```
Comment (wondersloth) [inline package.json:18]: Remove this.

File: package.json [hunk 2]
```diff
@@ -13,15 +13,17 @@
...
     "prepublishOnly": "scripts/build.js",
     "problems": "tsc -p tsconfig.json --noEmit",
     "start": "webpack-dev-server",
     "storybook": "yarn build && yarn workspace basic-example-app storybook",
-    "test": "yarn lint && testem ci && yarn test:babel-plugins",
+    "test": "yarn format:check && yarn lint && testem ci && yarn test:babel-plugins && yarn test:prettier",
```
Comment (wondersloth) [inline package.json:23]: We may not want this here.

File: packages/@glimmerx/core/tests/modifier-tests.ts [hunk 3]
```diff
@@ -7,7 +7,9 @@ import { renderComponent, didRender } from '..';
 module('Modifier Tests', () => {
   test('Supports the on modifier', async (assert) => {
     class MyComponent extends Component {
-      static template = hbs`<button {{on "click" this.incrementCounter}}>Count: {{this.count}}</button>`;
```
Comment (wondersloth) [inline packages/@glimmerx/core/tests/modifier-tests.ts:10]: This line hit the line length limit for the formatter, so it will wrap and indent.

**Review by wondersloth — COMMENTED** [review]
Preliminary review. Made some comments.

**Comment by wondersloth** [comment]
Need to add a test for an empty tagged template expression to ensure it prunes the whitespace.

**Review by chiragpat — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (wondersloth) [reply package.json:23]: I respond to this in a comment by @chiragpat below.

File: packages/@glimmerx/prettier-plugin-component-templates/index.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  return path.match(
+    (node) => {
+      return node.type === 'TemplateLiteral';
+    },
+    (node, name) => {
+      return (
+        node.type === 'TaggedTemplateExpression' &&
+        node.tag.type === 'Identifier' &&
+        node.tag.name === 'hbs' &&
+        name === 'quasi'
```
[the remark this answers is not in the document]
Comment (wondersloth) [reply packages/@glimmerx/prettier-plugin-component-templates/index.js:52]: I will create an issue for this.

File: package.json [hunk 5]
```diff
@@ -13,15 +13,17 @@
   ],
   "scripts": {
     "build": "scripts/build.js",
+    "format:check": "prettier --check 'packages/**/*.{js,ts}'",
```
[the remark this answers is not in the document]
Comment (wondersloth) [reply package.json:16]: When prettier and eslint both do formatting we get sometimes a double rewrite of a file, with the collision of rules being applied to a tagged template expression. 

What we're doing now is simply using prettier to do a style-check format and do a check then use eslint lint for anti-patterns.

File: packages/@glimmerx/prettier-plugin-component-templates/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (wondersloth) [reply packages/@glimmerx/prettier-plugin-component-templates/index.js:52]: Issue created: https://github.com/glimmerjs/glimmer-experimental/issues/121
